tests/TestCase/Controller/SentencesControllerTest.php

```diff
@@ -277,13 +278,12 @@ public function testEditAudio_asAdmin() {
         $this->assertRedirect('/jpn/sentences/show/1');
     }
 
-    public function testPaginateRedirectsPageOutOfBoundsToLastPage() {
+    public function testPaginateRedirectsPageOutOfBoundsToLastPage_asGuest() {
         $user = 'kazuki';
         $userId = 7;
-        $lastPage = 2;
-
+        $defaultNbPerPage = User::$defaultSettings['sentences_per_page'];
         $newSentences = array();
-        for ($i = 1; $i <= 100; $i++) {
+        for ($i = 1; $i <= rand(1,100); $i++) {
```

Comment:
Why did you add randomness here and on the other test? I don’t think it’s a good idea because it means we don’t run the exact same test every time everywhere. I may have a failing test while yours is passing, which can create a lot of confusion.